test/widgets/dialog_checks.dart

```diff
@@ -47,19 +77,39 @@ void checkNoErrorDialog(WidgetTester tester) {
   required String expectedMessage,
   String? expectedActionButtonText,
 }) {
-  final dialog = tester.widget<AlertDialog>(find.byType(AlertDialog));
-  tester.widget(find.descendant(matchRoot: true,
-    of: find.byWidget(dialog.title!), matching: find.text(expectedTitle)));
-  tester.widget(find.descendant(matchRoot: true,
-    of: find.byWidget(dialog.content!), matching: find.text(expectedMessage)));
+  switch (defaultTargetPlatform) {
+    case TargetPlatform.android:
+    case TargetPlatform.fuchsia:
+    case TargetPlatform.linux:
+    case TargetPlatform.windows:
+      final dialog = tester.widget<AlertDialog>(find.bySubtype<AlertDialog>());
+      tester.widget(find.descendant(matchRoot: true,
+        of: find.byWidget(dialog.title!), matching: find.text(expectedTitle)));
+      tester.widget(find.descendant(matchRoot: true,
+        of: find.byWidget(dialog.content!), matching: find.text(expectedMessage)));
+
+      final actionButton = tester.widget(find.descendant(of: find.byWidget(dialog),
+          matching: find.widgetWithText(TextButton, expectedActionButtonText ?? 'Continue')));
 
-  final actionButton = tester.widget(
-    find.descendant(of: find.byWidget(dialog),
-      matching: find.widgetWithText(TextButton, expectedActionButtonText ?? 'Continue')));
+      final cancelButton = tester.widget(find.descendant(of: find.byWidget(dialog),
+          matching: find.widgetWithText(TextButton, 'Cancel')));
 
-  final cancelButton = tester.widget(
-    find.descendant(of: find.byWidget(dialog),
-      matching: find.widgetWithText(TextButton, 'Cancel')));
+      return (actionButton, cancelButton);
 
-  return (actionButton, cancelButton);
+    case TargetPlatform.iOS:
+    case TargetPlatform.macOS:
+      final dialog = tester.widget<CupertinoAlertDialog>(find.byType(CupertinoAlertDialog));
+      tester.widget(find.descendant(matchRoot: true,
+        of: find.byWidget(dialog.title!), matching: find.text(expectedTitle)));
+      tester.widget(find.descendant(matchRoot: true,
+        of: find.byWidget(dialog.content!), matching: find.text(expectedMessage)));
+
+      final actionButton = tester.widget(find.descendant(of: find.byWidget(dialog),
+          matching: find.widgetWithText(CupertinoDialogAction, expectedActionButtonText ?? 'Continue')));
+
+      final cancelButton = tester.widget(find.descendant(of: find.byWidget(dialog),
+          matching: find.widgetWithText(CupertinoDialogAction, 'Cancel')));
+
+      return (actionButton, cancelButton);
```

Comment:
nit: group these as one stanza:
```suggestion
      final actionButton = tester.widget(find.descendant(of: find.byWidget(dialog),
          matching: find.widgetWithText(CupertinoDialogAction, expectedActionButtonText ?? 'Continue')));
      final cancelButton = tester.widget(find.descendant(of: find.byWidget(dialog),
          matching: find.widgetWithText(CupertinoDialogAction, 'Cancel')));
      return (actionButton, cancelButton);
```

That helps in seeing the overall organization, since there's now this extra layer of structure around these that is the `switch` statement.